a3p-integration/proposals/c:stake-bld/test-lib/rpc.js

```diff
@@ -0,0 +1,265 @@
+/** @file copied from packages/agoric-cli */
+// @ts-check
+/* global Buffer */
+
+import {
+  boardSlottingMarshaller,
+  makeBoardRemote,
+} from '@agoric/internal/src/marshal.js';
+import { Fail } from '@endo/errors';
+
+export { boardSlottingMarshaller };
+
+export const boardValToSlot = val => {
+  if ('getBoardId' in val) {
+    return val.getBoardId();
+  }
+  Fail`unknown obj in boardSlottingMarshaller.valToSlot ${val}`;
+};
+
+export const networkConfigUrl = agoricNetSubdomain =>
+  `https://${agoricNetSubdomain}.agoric.net/network-config`;
+export const rpcUrl = agoricNetSubdomain =>
+  `https://${agoricNetSubdomain}.rpc.agoric.net:443`;
+
+/**
+ * @typedef {{ rpcAddrs: string[], chainName: string }} MinimalNetworkConfig
+ */
+
+/** @type {MinimalNetworkConfig} */
+const networkConfig = {
+  rpcAddrs: ['http://0.0.0.0:26657'],
+  chainName: 'agoriclocal',
+};
+export { networkConfig };
+// console.warn('networkConfig', networkConfig);
+
+/**
+ * @param {object} powers
+ * @param {typeof window.fetch} powers.fetch
+ * @param {MinimalNetworkConfig} config
+ */
+export const makeVStorage = (powers, config = networkConfig) => {
+  /** @param {string} path */
+  const getJSON = path => {
+    const url = config.rpcAddrs[0] + path;
+    // console.warn('fetching', url);
+    return powers.fetch(url, { keepalive: true }).then(res => res.json());
+  };
+  // height=0 is the same as omitting height and implies the highest block
+  const url = (path = 'published', { kind = 'children', height = 0 } = {}) =>
+    `/abci_query?path=%22/custom/vstorage/${kind}/${path}%22&height=${height}`;
```

Comment:
Per [this comment](https://github.com/Agoric/agoric-sdk/pull/9078#discussion_r1529484514) I understand this code is from elsewhere. Not sure if in scope for this PR, but it's my understanding that `/abci_query` + /custom/vstorage` are sunsetting with upgrade-14.

Reply:
not in scope. this will be abstracted anyway in https://github.com/Agoric/agoric-sdk/issues/9109